clients/client-topia/src/factories/DroppedAssetFactory.ts

```diff
@@ -11,6 +12,25 @@ export class DroppedAssetFactory extends SDKController {
     return new DroppedAsset(this.topia, id, urlSlug, options);
   }
 
+  async createWithUniqueName(
+    uniqueName: string,
+    urlSlug: string,
+    interactivePublicKey: string,
+    interactiveSecret: string,
+  ): Promise<DroppedAsset> {
+    const interactiveJWT = jwt.sign(interactivePublicKey, interactiveSecret);
+    try {
+      const response: AxiosResponse = await this.topiaPublicApi().get(
+        `/world/${urlSlug}/asset-by-unique-name/${uniqueName}`,
```

Comment:
I think we need to remove the actual `get` from this.

we want to create the class object, but a dev potentially wants to utilize the class without having to await/fetch the stuff

ideally, a dev can create the class and get the data object directly without having to get the asset first. Will be much faster.

That brings up... let's get a way to handle get data object with just a unique name passed through

Reply:
@cpsiaki I'm a little confused by this. I thought the original request was for a dev to be able to pull all pertinent information for a dropped asset, including data object, by unique name. I'll definitely update the name of this to be `getWithUniqueName` as it more accurately describes what it's doing.
Separately, I could create a new method for `createWithUniqueName` that only creates the class object but that would be pretty useless because all of the methods in DroppedAsset (as well as the constructor) require the dropped asset's id in order to work.

Reply:
Yes. `createWithUniqueName` is what I'm looking for to get the data object without blocking by a `get` first.

Create the class, ask for the data object

This will require a new endpoint to to get data object by unique name

Reply:
overall the effort here is to speed up the retrieval of the information that a dev needs to progress a game forward.... right now its:

asset click
`await` getAssets
look for scoreboard
`await` get scoreboard asset
`await` get scoreboard asset data object

finally do something

So speeding that up into:

`await` get asset data object by unique name

do something